dev/integration_tests/android_views/android/app/src/main/java/io/flutter/integration/androidviews/MotionEventCodec.java

```diff
@@ -0,0 +1,126 @@
+import static android.view.MotionEvent.PointerProperties;
+
+@TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+public class MotionEventCodec {
+
+    public static HashMap<String, Object> encode(MotionEvent event) {
+        ArrayList<HashMap<String,Object>> pointerProperties = new ArrayList<>();
+        ArrayList<HashMap<String,Object>> pointerCoords = new ArrayList<>();
+
+        for (int i = 0; i < event.getPointerCount(); i++) {
+            MotionEvent.PointerProperties properties = new MotionEvent.PointerProperties();
+            event.getPointerProperties(i, properties);
+            pointerProperties.add(encodePointerProperties(properties));
+
+            MotionEvent.PointerCoords coords = new MotionEvent.PointerCoords();
+            event.getPointerCoords(i, coords);
+            pointerCoords.add(encodePointerCoords(coords));
+        }
+
+        HashMap<String, Object> eventMap = new HashMap<>();
+        eventMap.put("downTime", event.getDownTime());
+        eventMap.put("eventTime", event.getEventTime());
+        eventMap.put("action", event.getAction());
+        eventMap.put("pointerCount", event.getPointerCount());
+        eventMap.put("pointerProperties", pointerProperties);
+        eventMap.put("pointerCoords", pointerCoords);
+        eventMap.put("metaState", event.getMetaState());
+        eventMap.put("buttonState", event.getButtonState());
+        eventMap.put("xPrecision", event.getXPrecision());
+        eventMap.put("yPrecision", event.getYPrecision());
+        eventMap.put("deviceId", event.getDeviceId());
+        eventMap.put("edgeFlags", event.getEdgeFlags());
+        eventMap.put("source", event.getSource());
+        eventMap.put("flags", event.getFlags());
+
+        return eventMap;
+    }
+
+    private static HashMap<String, Object> encodePointerProperties(PointerProperties properties) {
+        HashMap<String, Object> map = new HashMap<>();
+        map.put("id", properties.id);
+        map.put("toolType", properties.toolType);
+        return map;
+    }
+
+    private static HashMap<String, Object> encodePointerCoords(PointerCoords coords) {
+        HashMap<String, Object> map = new HashMap<>();
+        map.put("orientation", coords.orientation);
+        map.put("pressure", coords.pressure);
+        map.put("size", coords.size);
+        map.put("toolMajor", coords.toolMajor);
+        map.put("toolMinor", coords.toolMinor);
+        map.put("touchMajor", coords.touchMajor);
+        map.put("touchMinor", coords.touchMinor);
+        map.put("x", coords.x);
+        map.put("y", coords.y);
+        return map;
+    }
+
+    @SuppressWarnings("unchecked")
+    public static MotionEvent decode(HashMap<String, Object> data) {
+        List<PointerProperties> pointerProperties = new ArrayList<>();
+        List<PointerCoords> pointerCoords = new ArrayList<>();
+
+        for (HashMap<String, Object> property : (List<HashMap<String, Object>>)data.get("pointerProperties")) {
```

Comment:
https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace `#5`, add a space after casting or be consistent in the file.

Reply:
done